keyless/pepper/service/src/lib.rs

```diff
@@ -129,6 +137,139 @@ impl HandlerTrait<PepperRequest, SignatureResponse> for V0SignatureHandler {
     }
 }
 
+#[macro_export]
+macro_rules! invalid_signature {
+    ($message:expr) => {
+        BadRequest($message.to_owned())
+    };
+}
+
+pub struct V0VerifyHandler;
+
+#[async_trait]
+impl HandlerTrait<VerifyRequest, VerifyResponse> for V0VerifyHandler {
+    async fn handle(&self, request: VerifyRequest) -> Result<VerifyResponse, ProcessingFailure> {
+        let VerifyRequest {
+            public_key,
+            signature,
+            message,
+            address: _,
+        } = request;
+        if let (AnyPublicKey::Keyless { public_key }, AnySignature::Keyless { signature }) =
+            (&public_key, &signature)
+        {
+            let KeylessPublicKey { idc: _, iss_val } = public_key;
+            let KeylessSignature {
+                cert,
+                jwt_header_json: _,
+                exp_date_secs: _,
+                ephemeral_pubkey,
+                ephemeral_signature,
+            } = signature;
+            ephemeral_signature
+                .verify_arbitrary_msg(&message, ephemeral_pubkey)
+                .map_err(|e| BadRequest(format!("Ephemeral sig check failed: {e}")))?;
+            let jwt_header = signature
+                .parse_jwt_header()
+                .map_err(|e| BadRequest(format!("JWT header decoding error: {e}")))?;
+            let jwk = jwk::cached_decoding_key_as_rsa(iss_val, &jwt_header.kid)
```

Comment:
oh this is the reason you refactored `DECODING_KEY_CACHE`?

Reply:
yes